compiler/ast/reports.nim

```diff
@@ -712,6 +712,7 @@ type
 
   Report* = object
     ## Toplevel wrapper type for the compiler report
+    reportId: ReportId
```

Comment:
Report should not store its ID directly, instead the ID to free should be passed to the cleanup logic.